### PR TITLE
Update the API docs for snapshots

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: addon/-private/system/snapshot.js [hunk 1]
```diff
@@ -347,6 +347,26 @@ Snapshot.prototype = {
...
+    Serializes the snapshot using the serializer for the model.
+
+    Example
+
+    ```app/adapters/application.js
+    import DS from 'ember-data';
+
+    export default DS.Adapter.extend({
+      createRecord: function(store, type, snapshot) {
+        var data = this.serialize(snapshot, { includeId: true });
```
Comment (sly7-7) [inline addon/-private/system/snapshot.js]: why not use snapshot.serialize ?

Comment (bmac) [inline addon/-private/system/snapshot.js]: Oops. I copied and pasted from the adapter docs and forgot to edit this line. Updated.

File: addon/-private/system/snapshot-record-array.js [hunk 2]
```diff
@@ -25,36 +25,125 @@ export default function SnapshotRecordArray(recordArray, meta, options = {}) {
...
   /**
     Number of records in the array
+
+    Example
+
+    ```app/adapters/post.js
+    import DS from 'ember-data'
+
+    export default DS.JSONAPIAdapter.extend({
+      shouldReloadAll: function(store, snapshotRecordArray) {
```
Comment (locks) [inline addon/-private/system/snapshot-record-array.js]: ES2015 method shorthand: `shouldReloadAll(store, snapshotRecordArray) {`.

File: addon/-private/system/snapshot-record-array.js [hunk 3]
```diff
@@ -25,36 +25,125 @@ export default function SnapshotRecordArray(recordArray, meta, options = {}) {
...
-    A hash of adapter options
+    A hash of adapter options passed into the store method for this request.
+
+    Example
+
+    ```app/adapters/post.js
+    import MyCustomAdapter from './custom-adapter';
+
+    export default MyCustomAdapter.extend({
+      findAll: function(store, type, sinceToken, snapshotRecordArray) {
```
Comment (locks) [inline addon/-private/system/snapshot-record-array.js]: ES2015 method shorthand: `findAll(store, type, sinceToken, snapshotRecordArray) {`.

File: addon/-private/system/snapshot-record-array.js [hunk 4]
```diff
@@ -25,36 +25,125 @@ export default function SnapshotRecordArray(recordArray, meta, options = {}) {
...
+  /**
+    The relationships to include for this request.
+
+    Example
+
+    ```app/adapters/application.js
+    import DS from 'ember-data';
+
+    export default DS.Adapter.extend({
+      findAll: function(store, type, snapshotRecordArray) {
```
Comment (locks) [inline addon/-private/system/snapshot-record-array.js]: ES2015 method shorthand: `findAll(store, type, snapshotRecordArray) {`.

File: addon/-private/system/snapshot-record-array.js [hunk 5]
```diff
@@ -25,36 +25,125 @@ export default function SnapshotRecordArray(recordArray, meta, options = {}) {
...
 /**
   Get snapshots of the underlying record array
+
+  Example
+
+  ```app/adapters/post.js
+  import DS from 'ember-data'
+
+  export default DS.JSONAPIAdapter.extend({
+    shouldReloadAll: function(store, snapshotArray) {
```
Comment (locks) [inline addon/-private/system/snapshot-record-array.js]: ES2015 method shorthand: `shouldReloadAll(store, snapshotArray) {`.

File: addon/-private/system/snapshot.js [hunk 6]
```diff
@@ -347,6 +347,26 @@ Snapshot.prototype = {
...
   /**
+    Serializes the snapshot using the serializer for the model.
+
+    Example
+
+    ```app/adapters/application.js
+    import DS from 'ember-data';
+
+    export default DS.Adapter.extend({
+      createRecord: function(store, type, snapshot) {
```
Comment (locks) [inline addon/-private/system/snapshot.js]: ES2015 method shorthand: `createRecord(store, type, snapshot) {`.

**Comment by bmac** [comment]
@locks did these changes address your issues?

**Pull request merged**